lib/crystalball/rails/map_generator/i18n_strategy/simple_patch.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'crystalball/map_generator/helpers/method_replacer'
+require 'active_support/core_ext/hash'
+
+module Crystalball
+  module Rails
+    class MapGenerator
+      class I18nStrategy
+        # Module to add new patched `load_file`, `store_translations` and `lookup`
+        # methods to `I18n::Backend::Simple`.
+        module SimplePatch
+          class << self
+            # Patches `I18n::Backend::Simple`.
+            def apply!
+              ::I18n::Backend::Simple.class_eval do
+                include SimplePatch
+                extend ::Crystalball::MapGenerator::Helpers::MethodReplacer
+
+                replace_with_patch :load_file, :store_translations, :lookup
+              end
+            end
+
+            # Reverts original behavior of `I18n::Backend::Simple`
+            def revert!
+              ::I18n::Backend::Simple.class_eval do
+                restore_to_original :load_file, :store_translations, :lookup
+              end
+            end
+          end
+
+          # Will replace original `I18n::Backend::Simple#load_file`.
+          # Passes filename to #store_translations in options
+          def cb_patched_load_file(filename)
+            type = File.extname(filename).tr('.', '').downcase
+            raise ::I18n::UnknownFileType.new(type, filename) unless respond_to?(:"load_#{type}", true)
+            data = send(:"load_#{type}", filename)
+            unless data.is_a?(Hash)
+              raise ::I18n::InvalidLocaleData.new(filename, 'expects it to return a hash, but does not')
+            end
+            data.each { |locale, d| store_translations(locale, d || {}, cb_filename: filename) } # only this line changed compared to original
+          end
+
+          # Will replace original `I18n::Backend::Simple#store_translations`.
+          # Adds filename for each value
+          def cb_patched_store_translations(locale, data, options = {}) # rubocop:disable Metrics/MethodLength
+            if I18n.enforce_available_locales &&
+               I18n.available_locales_initialized? &&
+               !I18n.available_locales.include?(locale.to_sym) &&
+               !I18n.available_locales.include?(locale.to_s)
+              return data
+            end
+            locale = locale.to_sym
+            translations[locale] ||= {}
+            data = data.deep_symbolize_keys
+            cb_add_filename_to_values(data, options[:cb_filename]) # only this line added compared to original
+            translations[locale].deep_merge!(data)
+          end
```

Comment:
It seems like these 2 patched files contain a lot of copy-pasted code from a gem. Lets pair on it tomorrow